pkg/zk/generators.go

```diff
@@ -32,8 +32,30 @@ func headlessSvcName(z *v1beta1.ZookeeperCluster) string {
 	return fmt.Sprintf("%s-headless", z.GetName())
 }
 
+var zkDataVolume = "data"
+
 // MakeStatefulSet return a zookeeper stateful set from the zk spec
 func MakeStatefulSet(z *v1beta1.ZookeeperCluster) *appsv1.StatefulSet {
+	extraVolumes := []v1.Volume{}
+	persistence := z.Spec.Persistence
+	pvcs := []v1.PersistentVolumeClaim{}
+	if persistence != nil {
+		pvcs = append(pvcs, v1.PersistentVolumeClaim{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:   zkDataVolume,
+				Labels: map[string]string{"app": z.GetName()},
+			},
+			Spec: persistence.PersistentVolumeClaimSpec,
+		})
+	} else {
+		extraVolumes = append(extraVolumes, v1.Volume{
```

Comment:
Our zookeeperStartup.sh script relies on **persistent volumes** to be able to handle server restarts which will be broken when using ephemeral volumes.